modules/accredited_representative_portal/app/models/accredited_representative_portal/claimant_representative.rb

```diff
@@ -26,21 +26,23 @@ def perform
           MSG
         end
 
-        poa_holder = @claimant.power_of_attorney_holder
-        poa_holders = @representative.power_of_attorney_holders
-
-        poa_holders.each do |h|
-          ##
-          # Might be nice to instead have a `PowerOfAttorneyHolder#==` with a
-          # semantics that matches what is needed here.
-          #
-          next unless h.poa_code == poa_holder.poa_code
-          next unless h.type == poa_holder.type
-
-          return build(h)
-        end
+        @claimant.power_of_attorney_holder.present? or
+          return nil
+
+        common_poa_holder =
+          @representative.power_of_attorney_holders.find do |h|
```

Comment:
Do you need more htan one match here?

If you want more than one match, I think you'd want to use select instead.